node/modules/storageminer.go

```diff
@@ -675,7 +676,7 @@ func RetrievalProvider(
 	dt dtypes.ProviderDataTransfer,
 	pricingFnc dtypes.RetrievalPricingFunc,
 	userFilter dtypes.RetrievalDealFilter,
-	dagStore mktdagstore.DagStoreWrapper,
+	dagStore shared.DagStoreWrapper,
```

Comment:
Let's avoid using _their_ interface here. Instead, just wire in _our_ concrete type `dagstore.Wrapper`, which will be present in the DI graph, and the `NewProvider` constructor will accept because it implements the interface _they_ want.

nit: I'm not a fan of shared/util/helpers/etc. package names in Go. It's actually [quite frowned upon](https://dave.cheney.net/2019/01/08/avoid-package-names-like-base-util-or-common). ¯\_(ツ)_/¯